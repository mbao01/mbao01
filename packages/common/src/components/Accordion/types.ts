import * as AccordionPrimitive from "@radix-ui/react-accordion";

export type AccordionProps = React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Root>;

export type AccordionContentProps = React.ComponentPropsWithoutRef<
  typeof AccordionPrimitive.Content
>;

export type AccordionItemProps = React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Item>;

export type AccordionTriggerProps = React.ComponentPropsWithoutRef<
  typeof AccordionPrimitive.Trigger
>;
