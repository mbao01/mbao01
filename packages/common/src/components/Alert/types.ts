import { type VariantProps } from "../../libs";
import { getAlertClasses } from "./constants";

export type AlertProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getAlertClasses>;

export type AlertTitleProps = React.HTMLAttributes<HTMLHeadingElement>;

export type AlertDescriptionProps = React.HTMLAttributes<HTMLDivElement>;
