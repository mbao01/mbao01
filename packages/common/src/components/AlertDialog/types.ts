import * as AlertDialogPrimitive from "@radix-ui/react-alert-dialog";
import { type VariantProps } from "../../libs";
import { getButtonClasses } from "../Button/constants";
import {
  getDialogContentClasses,
  getDialogDescriptionClasses,
  getDialogFooterClasses,
  getDialogHeaderClasses,
  getDialogOverlayClasses,
  getDialogTitleClasses,
} from "../Dialog/constants";

export type AlertDialogProps = React.ComponentPropsWithoutRef<typeof AlertDialogPrimitive.Root>;

export type AlertDialogOverlayProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Overlay
> &
  VariantProps<typeof getDialogOverlayClasses>;

export type AlertDialogContentProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Content
> &
  VariantProps<typeof getDialogContentClasses>;

export type AlertDialogHeaderProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getDialogHeaderClasses>;

export type AlertDialogFooterProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getDialogFooterClasses>;

export type AlertDialogTitleProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Title
> &
  VariantProps<typeof getDialogTitleClasses>;

export type AlertDialogDescriptionProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Description
> &
  VariantProps<typeof getDialogDescriptionClasses>;

export type AlertDialogActionProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Cancel
> &
  VariantProps<typeof getButtonClasses>;

export type AlertDialogCancelProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Cancel
> &
  VariantProps<typeof getButtonClasses>;
