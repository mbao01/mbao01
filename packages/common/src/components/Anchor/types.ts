import { VariantProps } from "../../libs";
import { getAnchorClasses } from "./constant";

export type AnchorProps = React.AnchorHTMLAttributes<HTMLAnchorElement> & {
  isExternal?: boolean;
} & VariantProps<typeof getAnchorClasses>;
