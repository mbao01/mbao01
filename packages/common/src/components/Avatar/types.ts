import * as AvatarPrimitive from "@radix-ui/react-avatar";
import { type VariantProps } from "../../libs";
import {
  getAvatarClasses,
  getAvatarFallbackClasses,
  getAvatarImageClasses,
} from "./constants";

export type AvatarProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Root>,
  "asChild" | "size"
> &
  VariantProps<typeof getAvatarClasses>;

export type AvatarImageProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Image>,
  "asChild"
> &
  VariantProps<typeof getAvatarImageClasses>;

export type AvatarFallbackProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Fallback>,
  "asChild"
> &
  VariantProps<typeof getAvatarFallbackClasses>;
