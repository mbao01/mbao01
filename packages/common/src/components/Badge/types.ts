import { type VariantProps } from "../../helpers";
import { getBadgeClasses } from "./constants";

export type BadgeProps = React.HTMLAttributes<HTMLSpanElement> &
  VariantProps<typeof getBadgeClasses>;
