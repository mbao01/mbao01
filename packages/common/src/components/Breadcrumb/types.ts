import { ReactNode } from "react";

export type BreadcrumbProps = React.HTMLAttributes<HTMLElement> & {
  separator?: ReactNode;
};

export type BreadcrumbListProps = React.HTMLAttributes<HTMLUListElement>;

export type BreadcrumbItemProps = React.HTMLAttributes<HTMLLIElement>;

export type BreadcrumbLinkProps = React.HTMLAttributes<HTMLAnchorElement> & {
  asChild?: boolean;
};

export type BreadcrumbPageProps = React.HTMLAttributes<HTMLSpanElement>;

export type BreadcrumbSeparatorProps = React.HTMLAttributes<HTMLLIElement>;

export type BreadcrumbEllipsisProps = React.HTMLAttributes<HTMLSpanElement>;
