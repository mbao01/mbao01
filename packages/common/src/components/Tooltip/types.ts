import * as TooltipPrimitive from "@radix-ui/react-tooltip";
import { type VariantProps } from "../../libs";
import { getTooltipArrowClasses, getTooltipContentClasses } from "./constants";

export type TooltipArrowProps = React.HTMLAttributes<
  React.ElementRef<typeof TooltipPrimitive.Arrow>
> &
  TooltipPrimitive.TooltipArrowProps &
  VariantProps<typeof getTooltipArrowClasses>;

export type TooltipContentProps = React.HTMLAttributes<
  React.ElementRef<typeof TooltipPrimitive.Content>
> &
  TooltipPrimitive.TooltipContentProps &
  VariantProps<typeof getTooltipContentClasses>;
