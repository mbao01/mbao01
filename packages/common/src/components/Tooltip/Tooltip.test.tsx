import { render, screen } from "@testing-library/react";
import { Tooltip } from "./";
import { type TooltipProps } from "./types";

describe("Tooltip", () => {
  const renderTooltip = ({
    content,
    variant,
    children,
    position,
  }: TooltipProps) =>
    render(
      <Tooltip content={content} variant={variant} position={position}>
        {children}
      </Tooltip>
    );
  it("shows tooltip with content", () => {
    const { asFragment } = renderTooltip({
      content: "I am the tooltip's content",
      children: "Tooltip trigger",
    });

    const tooltipEl = screen.getByText("Tooltip trigger");

    expect(tooltipEl).toBeVisible();
    expect(tooltipEl).toHaveClass("tooltip");
    expect(tooltipEl).toHaveAttribute("data-tip", "I am the tooltip's content");
    expect(asFragment()).toMatchSnapshot();
  });
});
