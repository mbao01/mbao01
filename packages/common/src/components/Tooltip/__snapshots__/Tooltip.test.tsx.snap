// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tooltip > shows tooltip with content 1`] = `
<DocumentFragment>
  <div
    class="tooltip"
    data-tip="I am the tooltip's content"
  >
    Tooltip trigger
  </div>
</DocumentFragment>
`;
